File: Python/POO I/Exercicios 22/Pescador.py
# -*- coding: utf-8 -*-
class Pescador:
    __total , __melhor = 0,None
    def __init__(self,nome,idade):
        self.__nome = nome
        self.__idade = idade

    def pescou(self):
        self.__total += 1

    def getNome(self):
        return self.__nome
    def getIdade(self):
        return self.__idade
    def getTotal(self):
        return self.__total

def melhor():
    if bob.getTotal() > ted.getTotal() and bob.getTotal() > tom.getTotal():
        return bob.getNome()
    elif ted.getTotal() > bob.getTotal() and ted.getTotal() > tom.getTotal():
        return ted.getNome()
    else:
        return tom.getNome()
def pior():
    if bob.getTotal() < ted.getTotal() and bob.getTotal() < tom.getTotal():
        return bob.getNome()
    elif ted.getTotal() < bob.getTotal() and ted.getTotal() < tom.getTotal():
        return ted.getNome()
    else:
        return tom.getNome()

#A
bob = Pescador('Bob',20)
#B
bob.pescou()
#C
ted = Pescador('Ted',22)
#D
ted.pescou()
#E
ted.pescou()
#F
bob.pescou()
#G
tom = Pescador('Tom',43)
#H
tom.pescou()
#I
ted.pescou()
#J
print('{} {} {}'.format(bob.getNome(),bob.getIdade(),bob.getTotal()))
print('{} {} {}'.format(ted.getNome(),ted.getIdade(),ted.getTotal()))
print('{} {} {}'.format(tom.getNome(),tom.getIdade(),tom.getTotal()))
#K e L
print()
print(melhor())
print(pior())

File: C++/Pescador.cpp
#include "Pescador.hpp"
#include <vector>

using namespace std;

int main(){
    vector<Pescador>lista;
    Pescador melhor("",0);
    Pescador pior("",0);

    Pescador bob("Bob",20);
    bob.pescou();

    Pescador ted("Ted",22);
    ted.pescou();
    ted.pescou();
    bob.pescou();

    Pescador tom("Tom",43);
    tom.pescou();
    ted.pescou();

    lista.push_back(bob);
    lista.push_back(ted);
    lista.push_back(tom);

    for(auto & value:lista){
        printf("Nome: %s\nIdade: %d\nPeixes: %d\n\n",value.getNome().c_str(),value.getIdade(),value.getPeixes());
        if (value.getPeixes() > melhor.getPeixes()){
            melhor = value;
        } if (pior.getNome() == "") {
            pior = value;
        } else {
            if(pior.getPeixes() > value.getPeixes()){
                pior = value;
            }
        }
    }

    printf("Melhor: %s\nPeixes: %d\n\n",melhor.getNome().c_str(),melhor.getPeixes());
    printf("Pior: %s\nPeixes: %d\n\n",pior.getNome().c_str(),pior.getPeixes());


    return 0;
}
